### PR TITLE
rook is migrating to lastest operator-sdk

### PR DESCRIPTION
**Pull request by subhamkrai**
this commit adapts rook new csv generation methomd.

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

**Comment by subhamkrai** [comment]
fyi @iamniting ^^

**Review by travisn — CHANGES_REQUESTED** [review]
The VolumeReplicationClasses and VolumeReplications CRDs still need to be generated. The previous csv generation had a special step for them since Rook doesn't own them, but we still want them to be in Rook's csv.

**Comment by subhamkrai** [comment]
> The VolumeReplicationClasses and VolumeReplications CRDs still need to be generated. The previous csv generation had a special step for them since Rook doesn't own them, but we still want them to be in Rook's csv.

The rook is not owning the VolumeReplication crds but let's hear from @iamniting

**Comment by iamniting** [comment]
> > The VolumeReplicationClasses and VolumeReplications CRDs still need to be generated. The previous csv generation had a special step for them since Rook doesn't own them, but we still want them to be in Rook's csv.
> 
> The rook is not owning the VolumeReplication crds but let's hear from @iamniting

Subham is right, We don't generate the VolumeReplication CRDs, They are owned by CSI addons nowadays.

**Comment by travisn** [comment]
> > > The VolumeReplicationClasses and VolumeReplications CRDs still need to be generated. The previous csv generation had a special step for them since Rook doesn't own them, but we still want them to be in Rook's csv.
> > 
> > 
> > The rook is not owning the VolumeReplication crds but let's hear from @iamniting
> 
> Subham is right, We don't generate the VolumeReplication CRDs, They are owned by CSI addons nowadays.

Correct, they are owned by CSI addons, but they have always been included in the Rook csv. If they are being removed from the Rook csv, they would need to be added elsewhere and confirm there is no issue with the change in ownership during the upgrades.

**Comment by iamniting** [comment]
> > > > The VolumeReplicationClasses and VolumeReplications CRDs still need to be generated. The previous csv generation had a special step for them since Rook doesn't own them, but we still want them to be in Rook's csv.
> > > 
> > > 
> > > The rook is not owning the VolumeReplication crds but let's hear from @iamniting
> > 
> > 
> > Subham is right, We don't generate the VolumeReplication CRDs, They are owned by CSI addons nowadays.
> 
> Correct, they are owned by CSI addons, but they have always been included in the Rook csv. If they are being removed from the Rook csv, they would need to be added elsewhere and confirm there is no issue with the change in ownership during the upgrades.

As far as I can see they have been dropped from the ocs csv in the 4.12 release itself and we did not hear any complaint from QE having a problem with the upgrades. So we should be good removing it. Adding @Madhu-1 to confirm it once.

**Comment by subhamkrai** [comment]
@iamniting @travisn PR is ready for final review, based on my changes on the rook, there was a successful build and the acceptance suite also passed with that.

cc @agarwal-mudit

**Review by iamniting — CHANGES_REQUESTED** [review]
Please also update the fusion bundle. Also, I believe we should use a different tag instead of master.

**Comment by subhamkrai** [comment]
> Please also update the fusion bundle. Also, I believe we should use a different tag instead of master.

I added fusion changes and I'll update rook tag once 1.11 is release

File: hack/common.sh
Comment (iamniting) [reply hack/common.sh:53]: Hi Travis, I would like to have 1.11 beta tag

File: deploy/csv-templates/crds/rook/ceph.rook.io_cephbucketnotifications.yaml
Comment (subhamkrai) [reply deploy/csv-templates/crds/rook/ceph.rook.io_cephbucketnotifications.yaml:20]: this, is coming from rook and for generating csv we remove the description in rook

Comment (travisn) [reply deploy/csv-templates/crds/rook/ceph.rook.io_cephbucketnotifications.yaml:20]: If we remove it, why is it showing up here?

Comment (subhamkrai) [reply deploy/csv-templates/crds/rook/ceph.rook.io_cephbucketnotifications.yaml:20]: right, https://github.com/rook/rook/pull/11724 this will fix this issue.

note: this issue will faced in upstream ocs-operator and not in downstream as there we directly use `make csv` and current problem was only in `make build`

Comment (subhamkrai) [reply deploy/csv-templates/crds/rook/ceph.rook.io_cephbucketnotifications.yaml]: see with https://github.com/red-hat-storage/ocs-operator/pull/1915/files#diff-f6f13ef4e3c83b5e22c436df5b1a4b24fd2c22a80d25855668b41828bcc05e12R53 this changes it is working, need to removing my tag once, included in rook release

File: hack/common.sh
Comment (subhamkrai) [inline hack/common.sh:53]:
```suggestion
LATEST_ROOK_IMAGE="subham03/csv:2"
```
need to use official tag

Comment (subhamkrai) [reply hack/common.sh]: reverted, good to go

**Review by travisn — COMMENTED** [review]
/lgtm

**Review by travisn — COMMENTED** [review]
/lgtm

**Review by iamniting — APPROVED** [review]

**Comment by agarwal-mudit** [comment]
Can someone confirm if the failure is not related to the changes? I would like to merge this because it is blocking a couple of other things. @subhamkrai @iamniting @malayparida2000

**Comment by iamniting** [comment]
> Can someone confirm if the failure is not related to the changes? I would like to merge this because it is blocking a couple of other things. @subhamkrai @iamniting @malayparida2000

We are trying to debug.

**Comment by iamniting** [comment]
> > Can someone confirm if the failure is not related to the changes? I would like to merge this because it is blocking a couple of other things. @subhamkrai @iamniting @malayparida2000
> 
> We are trying to debug.

Finally I am able to debug it via creating a dummy PR as I don't had the access to the cluster created in the subham PR.  I see that "csi-rbdplugin" is not running. The image "quay.io/cephcsi/cephcsi:v3.7.2" is being used for the "csi-rbdplugin," which does not have a "enable-read-affinity" flag defined, and rook is passing that flag to the container. To make it work, we need to update the latest "csi-rbdplugin" image in the rook or either we should not pass this argument.

cc @Rakshith-R 


```
$ oc get pods | grep csi-rbdplugin
csi-rbdplugin-bhpct                                               2/3     CrashLoopBackOff   10 (3m47s ago)   30m
csi-rbdplugin-provisioner-858d44c55d-2hsq7                        6/6     Running            0                30m
csi-rbdplugin-provisioner-858d44c55d-v6j5q                        6/6     Running            0                30m
csi-rbdplugin-rzbrz                                               2/3     CrashLoopBackOff   10 (3m31s ago)   30m
csi-rbdplugin-thj4l                                               2/3     CrashLoopBackOff   10 (3m55s ago)   30m
```

```
$ oc logs csi-rbdplugin-bhpct csi-rbdplugin
flag provided but not defined: -enable-read-affinity
```

```
  csi-rbdplugin:
    Container ID:  cri-o://30fe96c5b744f0378e47c9f778b573babbccca9e110e3bf64e1af8228f9e5239
    Image:         quay.io/cephcsi/cephcsi:v3.7.2
    Image ID:      quay.io/cephcsi/cephcsi@sha256:9dde057a7de88c4c1aab54ade1b0d85a79b0bca306d1be41488e59404f88fbf1
    Port:          <none>
    Host Port:     <none>
    Args:
      --nodeid=$(NODE_ID)
      --endpoint=$(CSI_ENDPOINT)
      --v=5
      --type=rbd
      --nodeserver=true
      --drivername=openshift-storage.rbd.csi.ceph.com
      --pidlimit=-1
      --metricsport=9090
      --metricspath=/metrics
      --enablegrpcmetrics=false
      --stagingpath=/var/lib/kubelet/plugins/kubernetes.io/csi/
      --csi-addons-endpoint=$(CSIADDONS_ENDPOINT)
      --enable-read-affinity=true
      --crush-location-labels=kubernetes.io/hostname,topology.kubernetes.io/region,topology.kubernetes.io/zone,topology.rook.io/chassis,topology.rook.io/rack,topology.rook.io/row,topology.rook.io/pdu,topology.rook.io/pod,topology.rook.io/room,topology.rook.io/datacenter
```

**Comment by Madhu-1** [comment]
Yes you need to wait for Rook 1.11, which carries cephcsi 3.8.0 release

**Comment by iamniting** [comment]
> Yes you need to wait for Rook 1.11, which carries cephcsi 3.8.0 release

@Madhu-1 It is blocking the DS we will use the beta tag to unblock it for now once your PR is merged.

**Comment by Madhu-1** [comment]
> > Yes you need to wait for Rook 1.11, which carries cephcsi 3.8.0 release
> 
> @Madhu-1 It is blocking the DS we will use the beta tag to unblock it for now once your PR is merged.

sounds good :+1:

**Comment by travisn** [comment]
The rook 1.11 branch build just finished to include ceph-csi 3.8. You can use tag `rook/ceph:v1.11.0-beta.0.38.gfa4d97ea6`.

**Comment by subhamkrai** [comment]
> The rook 1.11 branch build just finished to include ceph-csi 3.8. You can use tag `rook/ceph:v1.11.0-beta.0.38.gfa4d97ea6`.

yes, I'm using that. Thanks

**Comment by subhamkrai** [comment]
@iamniting I think `verify generate changes` test are flaky, some time it fails on 1.19 and then I push without any changes then it fails on 1.20. Also sometimes it fails on ocs csv change or ics csv changes. 
In my local I don't see any generation after running the commands.

```
003  IMAGE_BUILD_CMD=docker FUSION=true make gen-latest-csv 
 1004  git status 
 1005  IMAGE_BUILD_CMD=docker  make gen-latest-csv 
 1006  history 
~/go/src/github.com/ocs-operator
srai@192 ~ (use-rook-latest-csv-gen) $ git status 
On branch use-rook-latest-csv-gen
nothing to commit, working tree clean
```

```
Verifying latest CSV
hack/verify-latest-csv.sh
Success: no out of source tree changes found for CSV
```

**Comment by malayparida2000** [comment]
Hey @subhamkrai, Yeah this test is a bit flaky. I see the e2e tests have passed & I think we are good here.

**Comment by subhamkrai** [comment]
> Hey @subhamkrai, Yeah this test is a bit flaky. I see the e2e tests have passed & I think we are good here. Nitin will run those tests a few more times & it will pass.

thanks @malayparida2000 for confirmation.

**Comment by iamniting** [comment]
> @iamniting I think `verify generate changes` test are flaky, some time it fails on 1.19 and then I push without any changes then it fails on 1.20. Also sometimes it fails on ocs csv change or ics csv changes. In my local I don't see any generation after running the commands.
> 
> ```
> 003  IMAGE_BUILD_CMD=docker FUSION=true make gen-latest-csv 
>  1004  git status 
>  1005  IMAGE_BUILD_CMD=docker  make gen-latest-csv 
>  1006  history 
> ~/go/src/github.com/ocs-operator
> srai@192 ~ (use-rook-latest-csv-gen) $ git status 
> On branch use-rook-latest-csv-gen
> nothing to commit, working tree clean
> ```
> 
> ```
> Verifying latest CSV
> hack/verify-latest-csv.sh
> Success: no out of source tree changes found for CSV
> ```

It is passed now after re running it

**Review by iamniting — APPROVED** [review]

**Pull request merged**
